### PR TITLE
Added support for private repos and custom entrypoint

### PR DESCRIPTION
**Pull request by marko-gacesa**

### CODE REVIEW

File: engine/compiler/compiler.go [hunk 1]
```diff
@@ -54,6 +63,7 @@ func (c *Compiler) Compile(ctx context.Context, args runtime.CompilerArgs) runti
 	targetPool := pipeline.Pool.Use
 	pool := c.PoolManager.Get(targetPool)
 	if pool == nil {
+		// TODO return an error to the caller if the requested pool does not exist.
```
Comment (eoinmcafee00) [inline engine/compiler/compiler.go:66]: Is this still to do?

File: engine/compiler/compiler.go [hunk 2]
```diff
@@ -273,24 +291,58 @@ func (c *Compiler) Compile(ctx context.Context, args runtime.CompilerArgs) runti
...
 		})
 	}
 
+	// get registry credentials from registry plugins
+	creds, err := c.Registry.List(ctx, &registry.Request{
+		Repo:  args.Repo,
+		Build: args.Build,
+	})
+	if err != nil { //nolint:staticcheck
+		// TODO (bradrydzewski) return an error to the caller if the provider returns an error.
```
Comment (eoinmcafee00) [inline engine/compiler/compiler.go:319]: Doesn't seem much point to this statement?

**Review by eoinmcafee00 — CHANGES_REQUESTED** [review]
Just a couple of comments

File: engine/compiler/compiler.go (hunk 2 above)
Comment (marko-gacesa) [reply engine/compiler/compiler.go:319]: See the comment above... This particular code section is copied from the docker runner along with the "to do" comment.

**Comment by marko-gacesa** [comment]
The "to do"s must stay for now.

**Review by eoinmcafee00 — APPROVED** [review]

**Pull request merged**